src/components/Calendar.tsx

```diff
@@ -0,0 +1,166 @@
+                    { week.map((day, index) => (
+                        <div key={ day } style={{
+                            alignItems: 'center',
+                            display:'flex',
+                            flexDirection: 'column',
+                            gap:'1rem',
+                            height: '100%',
+                            justifyContent: 'start',
+                            width: '100%',
+                        }}>
+                            <Heading level={ 3 }>
+                                <Msg id={ day }/>
+                            </Heading>
+                            <Text data-testid={ day }>
+                                { new Date(new Date(monday).setDate(monday.getDate() + index)).getDate() }
+                            </Text>
+                        </div>
+                    )) }
+                </div>
+                <div ref={ calendar } style={{
+                    alignItems: 'center',
+                    display:'flex',
+                    gap:'0.5rem',
+                    height: '100rem',
+                    justifyContent: 'start',
+                    width: '100%',
+                }}>
+                    { week.map((day, index) => (
+                        <div key={ day } style={{
+                            display: 'flex',
+                            flexDirection: 'column',
+                            gap: '1rem',
+                            height: '100%',
+                            justifyContent: 'space-between',
+                            padding: '0',
+                            width: '100%',
+                        }}>
+                            <ul data-testid={ `${day}-events` } style={{
+                                background:'lightgray',
+                                height: '100%',
+                                listStyle: 'none',
+                                margin: 0,
+                                padding:0 ,
+                                position: 'relative',
+                                width: '100%',
+                            }}>
+                                { getEventsOfTheDay(index + 1)?.map(event => (
+                                    <CalendarEvent
+                                        key={ event.id }
+                                        endTime={ event.end_time }
+                                        id={ event.id }
+                                        startTime={ event.start_time }
+                                    />
+                                )) }
+                            </ul>
+                        </div>
+                    )) }
+                </div>
+            </div>
+        </>
+    );
+};
+
+interface CalendarEventProps {
+    startTime: string;
+    endTime: string;
+    id: number;
+}
+
+const CalendarEvent = ({ startTime, endTime, id }: CalendarEventProps): JSX.Element => {
+
+    const getPos = (start: string, end: string) => {
```

Comment:
Why is this a function that needs to be invoked twice in this component? Couldn't it just be code inline in the component that results in a `height` and a `top` constant that can then be passed to the style properties in the JSX?